server/app/forms/translation/ProgramTranslationForm.java

```diff
@@ -1,40 +1,153 @@
 package forms.translation;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
 import java.util.Locale;
+import java.util.Optional;
+import java.util.stream.IntStream;
+import play.data.DynamicForm;
+import play.data.FormFactory;
+import play.i18n.Lang;
+import play.libs.typedmap.TypedMap;
+import play.mvc.Http;
+import services.LocalizedStrings;
+import services.program.LocalizationUpdate;
 import services.program.ProgramDefinition;
+import services.program.StatusDefinitions;
+
+/**
+ * Form for updating translations for programs. This isn't a typical Play form in that the number of
+ * fields is dynamic (based on the collection of {@link StatusDefinitions.Status} associated with
+ * the program). Rather than binding using formFactory.form(ProgramTranslationForm.class), use
+ * ProgramTranslationForm.bindFromRequest()
+ */
+public final class ProgramTranslationForm {
+
+  public static final String DISPLAY_NAME_FORM_NAME = "displayName";
+  public static final String DISPLAY_DESCRIPTION_FORM_NAME = "displayDescription";
+
+  private final DynamicForm form;
+  private final int maxStatusTranslations;
+
+  private static final Lang DEFAULT_LANG = new Lang(LocalizedStrings.DEFAULT_LOCALE);
```

Comment:
move up with the other consts.

sending this on its own as GH is being weird and curious if/where it shows up after sending.

Reply:
ah it shows now!